src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -83,6 +83,11 @@ double RNN<OutputLayerType, InitializationRuleType, CustomLayers...>::Train(
     ResetParameters();
   }
 
+  if(rho>responses.size())
+  {
+    rho=responses.size();
```

Comment:
Looks like this introduces some issue, let me know if I should take a closer look into the issue.

Reply:
Yes at that moment I was stuck but after using gdb multiple I was able to understand the problem.